src/Annotations/Serializer/AnnotationNormalizer.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace eLife\Annotations\Serializer;
+
+use eLife\ApiSdk\ApiSdk;
+use eLife\HypothesisClient\Model\Annotation;
+use League\CommonMark\Block\Element;
+use League\CommonMark\DocParser;
+use League\CommonMark\ElementRendererInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareTrait;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+final class AnnotationNormalizer implements NormalizerInterface, NormalizerAwareInterface
+{
+    const CANNOT_RENDER_CONTENT_COPY = 'NOTE: It is not possible to display this content.';
+
+    use NormalizerAwareTrait;
+
+    private $docParser;
+    private $htmlRenderer;
+    private $logger;
+
+    public function __construct(DocParser $docParser, ElementRendererInterface $htmlRenderer, LoggerInterface $logger)
+    {
+        $this->docParser = $docParser;
+        $this->htmlRenderer = $htmlRenderer;
+
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param Annotation $object
+     */
+    public function normalize($object, $format = null, array $context = []) : array
+    {
+        $content = $object->getText() ? $this->processText($object->getText()) : null;
+
+        $data = array_filter([
```

Comment:
`array_filter` without a callback removes values that are `false`. But which of these values can be `false`? If it's only `parents` it can be inserted separately into the already created `$data`

Reply:
Will refactor.